examples/BooksellerFancy.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE BlockArguments #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+{-
+# Maximally fancy bookseller
+-}
+
+module BooksellerFancy where
+
+import CLI
+import Choreography
+import Choreography.Network.Http
+import Data (deliveryDateOf, priceOf)
+import System.Environment
+
+$(mkLoc "buyer")
+$(mkLoc "seller")
+
+proof1 :: Subset ("buyer" ': supporters) ("buyer" ': "seller" ': supporters)
```

Comment:
I think our convention should be that proofs' names reflect their use as _identifiers_, so I'd call these `buyers` (or `customers`?) and maybe `transactors`. Those aren't great names though...

Reply:
Oh, also note that `supporters` here is free; it's a complely different type-variable from the one in `bookseller`. These should probably be declared as `let`s or `where`s in `bookseller` (but I think they will _work_ here).